packages/@sanity/types/src/schema/definition/schemaDefinition.ts

```diff
@@ -0,0 +1,131 @@
+import {PreviewConfig} from '../preview'
+import {InitialValueProperty, SchemaValidationValue} from '../types'
+import {
+  ArrayDefinition,
+  BlockDefinition,
+  BooleanDefinition,
+  CrossDatasetReferenceDefinition,
+  DateDefinition,
+  DatetimeDefinition,
+  DocumentDefinition,
+  FileDefinition,
+  GeopointDefinition,
+  ImageDefinition,
+  NumberDefinition,
+  ObjectDefinition,
+  ReferenceDefinition,
+  SlugDefinition,
+  SpanDefinition,
+  StringDefinition,
+  TextDefinition,
+  UrlDefinition,
+} from './type'
+import {BaseDefinitionOptions} from './type/common'
+
+/**
+ * This type can be extended being a interface and not a type, via declaration merging
+ */
+export interface IntrinsicTypeDefinition {
+  array: ArrayDefinition
+  block: BlockDefinition
+  boolean: BooleanDefinition
+  date: DateDefinition
+  datetime: DatetimeDefinition
+  document: DocumentDefinition
+  file: FileDefinition
+  geopoint: GeopointDefinition
+  image: ImageDefinition
+  number: NumberDefinition
+  object: ObjectDefinition
+  reference: ReferenceDefinition
+  crossDatasetReference: CrossDatasetReferenceDefinition
+  slug: SlugDefinition
+  span: SpanDefinition
+  string: StringDefinition
+  text: TextDefinition
+  url: UrlDefinition
+}
+
+/**
+ * A union of all intrinsic types allowed natively in the schema.
+ */
+export type TypeName = IntrinsicTypeDefinition[keyof IntrinsicTypeDefinition]['type']
```

Comment:
`IntrinsicDefinitionTypeName`?

Reply:
I would really like to keep this short, as it is used in a lot of generic contexts. If it is a long name, that reads worse imo.

IntrinsicTypeName _perhaps_?

Reply:
I think `IntrinsicTypeName` makes sense, since it's built from keys of `IntrinsicTypeDefinition`.